<?xml version="1.0"?>
<launch>
    <!-- Ultrasonik sensör node'u -->
    <node pkg="barlas" type="ultrasonic_node.py" name="ultrasonic_node" output="screen">
        <param name="min_distance" value="0.02" />
        <!-- 2cm -->
        <param name="max_distance" value="4.0" />
        <!-- 4m -->
    </node>

    <!-- LIDAR node'u -->
    <node pkg="barlas" type="lidar_node.py" name="lidar_node" output="screen">
        <param name="port" value="/dev/ttyUSB0" />
        <param name="frame_id" value="laser" />
        <param name="scan_frequency" value="8" />
        <!-- Hz -->
        <param name="sample_rate" value="9" />
        <!-- kHz -->
    </node>

    <!-- Static Transform: robot_base -> laser -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="laser_broadcaster" args="0.1 0 0.1 0 0 0 robot_base laser" />

    <!-- RViz Konfigürasyonu -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find barlas)/rviz/sensors.rviz" required="false" />

    <!-- rqt_plot için bazı yararlı topic'ler -->
    <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot" args="/barlas/sensors/ultrasonic/distances/data[0:7]" required="false" />
</launch>
